### PR TITLE
PN-9242-e-2-e-mittente-invio-notifica-a-destinatario-con-indirizzo-fornito-dalla-pa

### PR DESCRIPTION
**Pull request by afus0**
## Short description
There isn't a test related to the **PN-9242** as you can see in the [confluence page](https://pagopa.atlassian.net/wiki/spaces/PN/pages/868778282/Mittente+-+Mapping+test-scenari)

## List of changes proposed in this pull request
- added new test file `128_A1_invioNotificaADestinatarioConIndirizzoFornitoDallaPA.feature`
- added new step file class `WebTool` with new utility methods:
```
- switchToPortal()
- openNewTab()
- closeTab()
```
- moved methods: `generatePaProtocolNumber(), decodeNotificationRequestId() and waitTime()` into `WebTool`
- added new feature steps into `NotificaMittentePagoPATest` such as: 
```
- Then Nella section Informazioni preliminari si inseriscono i dati della notifica
- Then Nella section Destinatario si inseriscono i dati del destinatario
- And Nella section Destinitario si clicca su {string} e si inseriscono i dati
- Then Nella section Allegati si carica un atto
- And Si verifica che la notifica è stata creata correttamente
- Then In parallelo si effettua l'accesso al portale destinatario persona fisica e si apre la notifica ricevuta
- Si verifica che la notifica abbia lo stato {string}
```
- added related methods to the new feature steps
- moved `utility` package into the main project folder
- added 2 new enum that manage the login with the token

## How to test
Simply run `mvn clean test -Dcucumber.filter.tags=@TA_invioNotificaConIndirizzoFornitoDallaPA` and verify if the test works fine.

- [x] Does this PR require a manual test in AWS CodeBuild?

### CODE REVIEW

**Comment by afus0** [comment]
[2 x AWS Test success! ](https://eu-south-1.console.aws.amazon.com/codesuite/codebuild/151559006927/projects/pn-RunFeE2eTestsCodeBuildProject/build/pn-RunFeE2eTestsCodeBuildProject%3Ab0ecb118-9653-47cf-929e-72e7db135cf4/reports?region=eu-south-1&reports-meta=eyJmIjp7InRleHQiOiIifSwicyI6eyJwcm9wZXJ0eSI6InN0YXR1cyJ9LCJuIjoyMCwiaSI6MH0)

File: src/main/java/it/pn/frontend/e2e/common/BasePage.java [hunk 1]
```diff
@@ -98,5 +110,7 @@ public void clearWebElementField(WebElement element) {
             element.sendKeys(Keys.BACK_SPACE);
         }
     }
+
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/common/BasePage.java]: delete extra /n

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/NotificaMittentePagoPATest.java [hunk 2]
```diff
@@ -923,6 +919,133 @@ public void siCliccaSullaSecondaPaginaDelleNotifiche(int pagina) {
...
+                logger.info("Tentativo n. " + maximumRetry + " - Stato notifica: " + statusNotifica);
+                maximumRetry++;
+            } while (statusNotifica.equals("WAITING"));
+            driver.navigate().refresh();
+            logger.info("La notifica è stata creata correttamente");
+        } else {
+            logger.error("NotificationRequestId non trovato, il codice della risposta al url /delivery/v2.3/requests è diverso di 202 ");
+            Assert.fail("NotificationRequestId non trovato, il codice della risposta al url /delivery/v2.3/requests è diverso di 202 ");
+        }
+    }
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/NotificaMittentePagoPATest.java]: you should write logic step in dedicated page and not in test file, you told me

File: src/test/java/it/pn/frontend/e2e/utility/WebTool.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+public class WebTool {
+    private static final Logger logger = LoggerFactory.getLogger("WebTool");
+    private static final WebDriver driver = Hooks.driver;
+    private final List<NetWorkInfo> netWorkInfos = Hooks.netWorkInfos;
+
+    // create custom enum
+    public enum AppPortal {
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/utility/WebTool.java]: insert this in enum folder and create dedicated file

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/NotificaMittentePagoPATest.java (hunk 2 above)
Comment (afus0) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/NotificaMittentePagoPATest.java]: That is true, in fact i add a comment to the step:
https://github.com/pagopa/pn-frontend-e2e/blob/c95c0d8ae40e3d3506c7371c0e01e01e6f709724/src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/NotificaMittentePagoPATest.java#L995-L998
I'll try to find a better solution, right now the `networkInfo` cant be imported outside the `/test` folder

**Review by leleOFA — APPROVED** [review]

**Pull request merged**